src/odemis/driver/actuator.py

```diff
@@ -2179,3 +2182,601 @@ def terminate(self):
+        self._metadata[model.MD_FAV_POS_DEACTIVE] = {'z': focus_deactive_value}
+        logging.info("%s changed to %s", model.MD_FAV_POS_DEACTIVE, self._metadata[model.MD_FAV_POS_DEACTIVE])
+
+        # Set the initial focus active position with range 0
+        self._metadata[model.MD_FAV_POS_ACTIVE] = {'z': self._range[0]}
+        logging.info("%s changed to %s", model.MD_FAV_POS_ACTIVE, self._metadata[model.MD_FAV_POS_ACTIVE])
+
+        self.referenced = model.VigilantAttribute({}, readonly=True)
+        # listen to reference changes from the underlying stage
+        logging.debug("Subscribing to reference of dependency %s", self._lensz.name)
+        self._lensz.referenced.subscribe(self._ondepStageReferenced)
+        self._ondepStageReferenced(self._lensz.referenced.value)
+
+    def _ondepStageReferenced(self, cref):
+        """
+        Update the referenced value from the underlying stage
+        """
+        # Directly set the value from the updated argument
+        if 'z' in cref:
+            self.referenced._set_value(cref['z'], force_write=True)
+
+    def updateMetadata(self, md):
+        if model.MD_FAV_POS_ACTIVE in md or model.MD_FAV_POS_DEACTIVE in md:
+            raise ValueError("%s and %s cannot be set manually" % model.MD_FAV_POS_ACTIVE % model.MD_FAV_POS_DEACTIVE)
+
+        super(LinkedHeightFocus, self).updateMetadata(md)
+        # self._metadata.update(md)
+        if model.MD_POS_COR in md:
+            self._updatePosition(self._lensz.position.value)
+
+    def _updatePosition(self, lens_pos):
+        """
+        update the position VA from the underlying dependency
+        (the method is public to be used from the parent)
+        """
+        # Only update position when MD_POS_COR is already configured
+        if model.MD_POS_COR not in self._metadata:
+            logging.error("%s not in metadata" % model.MD_POS_COR)
+            return
+        focus_val = self._getFocusValue(target_lensz=lens_pos['z'])
+        logging.debug("Reporting focus position %s", focus_val)
+        self.position._set_value({'z': focus_val}, force_write=True)
+
+        # If the new position is in focus range, set the MD_FAV_POS_ACTIVE with this new value
+        if self._isInRange():
+            self._metadata[model.MD_FAV_POS_ACTIVE] = {'z': self.position.value['z']}
+            logging.info("%s changed to %s", model.MD_FAV_POS_ACTIVE, self._metadata[model.MD_FAV_POS_ACTIVE])
+
+    def _isInRange(self, pos=None):
+        """
+        A helper function to check if current position is in focus range
+        :return: True if position in focus range, None otherwise
+        """
+        pos = self.position.value['z'] if pos is None else pos
+        # Add 1% margin for hardware slight errors
+        margin = (self._range[1] - self._range[0]) * 0.01
+        if (self._range[0] - margin) <= pos <= (self._range[1] + margin):
+            return True
```

Comment:
You can write `return self._range[0] - margin) <= pos <= (self._range[1] + margin)`